File: 01_Iniciante/1040_Media3.cpp
//https://www.beecrowd.com.br/judge/pt/problems/view/1040
#include <iostream>

using namespace std;

float n1,n2,n3,n4,n5 = 0.0;

int main()
{

    cin >> n1 >> n2 >> n3 >> n4;

    float soma_total = (n1 * 2) + (n2 * 3) + (n3 * 4) + (n4 * 1);
    float soma_peso = (2 + 3 + 4 + 1);
    float media =  soma_total / soma_peso;
    cout << fixed ;
    cout.precision(1);
    cout << "Media: " << media << endl;

    if (media >= 7.0) {
        cout << "Aluno aprovado." << endl;
    }
    else if (media < 5.0) {
        cout << "Aluno reprovado." << endl;
    }else {
        cout << "Aluno em exame." << endl;
        cin >> n5;

        cout << "Nota do exame: " << n5 << endl;

        media = (media + n5)/2;

        if (media >= 5.0)
        {
            cout << "Aluno aprovado." << endl;
            cout << "Media final: " << media << endl;
        }
        else{
            cout << "Aluno reprovado" << endl;
            cout << "Media final: " << media << endl;
        }
    }

    return 0;
}

File: 01_Iniciante/1040_Media 3.py
#Link da Questão -> https://www.urionlinejudge.com.br/judge/pt/problems/view/1040

entrada = input()
valor = entrada.split()

key = False
peso_valor = [2,3,4,1]
media = 0.0
peso = 0.0
i = 0

while i < 4:
    valor[i] = float(valor[i])
    media += (valor[i] * peso_valor[i])
    peso += peso_valor[i]
    i += 1

media = (media)/peso

print("Media: {:.1f}".format(media))

if media >= 7.0:
    print("Aluno aprovado.")
elif media < 5.0:
    print("Aluno reprovado.")
else:
    print("Aluno em exame.")
    key = True

if key:
    nota = input()
    media = (media + float(nota))/2
    print("Nota do exame: {}".format(nota))
    if media >= 5.0:
        print("Aluno aprovado.")
    else:
        print("Aluno reprovado.")

    print("Media final: {:.1f}".format(media))


